devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -1602,6 +2821,40 @@ private def getStatusForDescription(changeType) {
 	return ""
 }
 
+/**
+ * Parse the time using SimpleDateFormat
+ *
+ * @param time: The time in HH:mm format
+ *
+ * @return The map with hour and minute
+ */
+private def getWeekDayHoursAndMinutes(time) {
```

Comment:
if the time string is formatted as "HH:mm", it's much simpler to just do:

`time = time.split(':')
return [hour: time[0] as Integer , minute: time[1] as Integer]`